scenario-runner/my_scenario.py

```diff
@@ -0,0 +1,137 @@
+        :param bool criteria_enable:
+        :param float timeout: Threshold (in seconds) after which test automatically fails
+
+        :return: None
+        :rtype: None
+        """
+        # Must be defined before super() call because BasicScenario
+        # references is in its __init__() function.
+        self.timeout = timeout
+
+        super(MyScenario, self).__init__(
+            "MyScenario",
+            ego_vehicles,
+            config,
+            world,
+            debug_mode=debug_mode,
+            criteria_enable=criteria_enable,
+        )
+
+        self.world_map = CarlaDataProvider.get_map()
+
+        self.carma_vehicle = ego_vehicles[0]
+        self.other_actors_dict = {}
+
+    def _initialize_actors(self, config: ScenarioConfiguration) -> None:
+        """
+        Note: this function overrides the one in BasicScenario (parent
+        class), so this override is responsible for adding the actors
+        defined in the scenario XML configuration.
+
+        :param ScenarioConfiguration config: Specifications from
+        scenario XML configuration
+        :return: None
+        """
+        actors = CarlaDataProvider.request_new_actors(config.other_actors)
+
+        self.other_actors_dict = {
+            actor_config.rolename: actor
+            for actor_config, actor in zip(config.other_actors, actors)
+        }
+
+    def _setup_scenario_trigger(self, _: ScenarioConfiguration) -> None:
+        """
+        Set up the scenario start trigger
+
+        Note: this function overrides the abstract one in the
+        BasicScenario parent class. The base class's implementation adds
+        a trigger that prevents the scenario from starting until the
+        ego vehicle drives some distance. We don't want that trigger
+        for this scenario.
+
+        :return: None
+        """
+        pass
+
+    def _create_behavior(self):
+        """
+        Setup the behavior for NewScenario
+
+        Note: this function overrides the abstract one in the
+        BasicScenario parent class.
+
+        :return: Behavior tree root
+        """
+        start_condition = Idle(5, name="start_condition")
+
+        crossing_person = self.other_actors_dict["crossing_person"]
+        walk_across_street = KeepVelocity(
+            crossing_person, 2.0, 8.0, name="walk_across_street"
+        )
+
+        # end_condition = DriveDistance(carma_vehicle, 10)
+
+        root = py_trees.composites.Sequence(name="root_sequence")
+        root.add_child(start_condition)
+        root.add_child(walk_across_street)
+        # root.add_child(end_condition)
+
+        return root
+
+    def _create_test_criteria(self) -> list:
+        """
+        Setup the evaluation criteria for NewScenario
+
+        Note: this function overrides the one in BasicScenario (parent class).
+
+        :return: List of test criteria
+        """
+        return [
+            # CollisionTest(self.carma_vehicle)
```

Comment:
I'm inclined to leave commented-out code in since this is supposed to be an example/reference file. I can at least add a description comment above explaining how the line should be used.

Reply:
Thats fine.